### PR TITLE
python3Packages.peewee: 3.14.4 -> 3.14.8

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 3.14.8

Change log:
- https://github.com/coleifer/peewee/releases/tag/3.14.8
- https://github.com/coleifer/peewee/releases/tag/3.14.7
- https://github.com/coleifer/peewee/releases/tag/3.14.6
- https://github.com/coleifer/peewee/releases/tag/3.14.5

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/peewee/default.nix [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
   buildInputs = [
     sqlite
-    cython # compile speedups
+    cython
   ];
 
   propagatedBuildInputs = [
-    apsw # sqlite performance improvement
+    apsw
   ] ++ (lib.optional withPostgres psycopg2)
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/peewee/default.nix:34]:
```suggestion
  ] ++ lib.optional withPostgres psycopg2
```

File: pkgs/development/python-modules/peewee/default.nix [hunk 2]
```diff
@@ -1,50 +1,58 @@
...
-    ++ (lib.optional withMysql mysql-connector);
+  ++ (lib.optional withMysql mysql-connector);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/peewee/default.nix:35]:
```suggestion
  ++ lib.optional withMysql mysql-connector;
```

File: pkgs/development/python-modules/peewee/default.nix [hunk 3]
```diff
@@ -1,50 +1,58 @@
...
+
+  checkInputs = [
+    flask
+  ];
 
   doCheck = withPostgres;
 
+  checkPhase = ''
+    rm -r playhouse # avoid using the folder in the cwd
+    python runtests.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/peewee/default.nix:45]:
```suggestion
    ${python.interpreter} runtests.py
```

File: pkgs/development/python-modules/peewee/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/peewee/default.nix]: Thanks

File: pkgs/development/python-modules/peewee/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/peewee/default.nix]: Thanks

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 145179` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>7 packages failed to build and are new build failures:</summary>
  <ul>
<li>dyndnsc: log was empty</li>
<li>python39Packages.dependency-injector: log was empty</li>
<li>python39Packages.fastapi: log was empty</li>
<li>python39Packages.json-logging: log was empty</li>
<li>python39Packages.slowapi: log was empty</li>
<li>soco-cli: log was empty</li>
<li>theharvester: log was empty</li>
  </ul>
</details>
<details>
<summary>
29 packages built:
</summary>
<ul>
<li>buku</li>
<li>cozy</li>
<li>etesync-dav</li>
<li>gkraken</li>
<li>gwe</li>
<li>matrix-commander</li>
<li>mirage-im</li>
<li>opsdroid</li>
<li>pantalaimon</li>
<li>python38Packages.dependency-injector</li>
<li>python38Packages.etesync</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.flask-admin</li>
<li>python38Packages.json-logging</li>
<li>python38Packages.matrix-nio</li>
<li>python38Packages.nplusone</li>
<li>python38Packages.peewee</li>
<li>python38Packages.slowapi</li>
<li>python38Packages.wtf-peewee</li>
<li>python39Packages.etesync</li>
<li>python39Packages.flask-admin</li>
<li>python39Packages.matrix-nio</li>
<li>python39Packages.nplusone</li>
<li>python39Packages.peewee</li>
<li>python39Packages.wtf-peewee</li>
<li>sqlite-web</li>
<li>sublime-music</li>
<li>vorta</li>
<li>weechatScripts.weechat-matrix</li>
</ul>
</details>

**Pull request merged**